src/test/kotlin/org/austral/librawallet/account/AccountTransactionControllerTests.kt

```diff
@@ -0,0 +1,253 @@
+
+    @Autowired
+    private lateinit var databaseInitializationService: DatabaseInitializationService
+
+    @Autowired
+    lateinit var mockMvc: MockMvc
+
+    @Autowired
+    lateinit var userRepository: UserRepository
+
+    @Autowired
+    lateinit var accountRepository: AccountRepository
+
+    @Autowired
+    lateinit var transactionRepository: TransactionRepository
+
+    @Autowired
+    lateinit var passwordEncoder: PasswordEncoder
+
+    @Autowired
+    lateinit var jwtUtils: JwtUtil
+
+    private val transactionsUrl = "/api/accounts/{accountId}/transactions"
+
+    @BeforeEach
+    fun setup() {
+        databaseInitializationService.clean()
+    }
+
+    private fun createUserAndToken(email: String, password: String): Pair<User, String> {
+        val hashed = passwordEncoder.encode(password)
+        val user = userRepository.save(User(email = email, password = hashed))
+        val token = "Bearer ${jwtUtils.generateToken(user)}"
+        return user to token
+    }
+
+    private fun createTransactions(account: Account, otherAccount: Account) {
+        val income = Transaction(
+            account = account,
+            otherAccount = otherAccount,
+            type = TransactionType.INCOME,
+            amount = 5000L, // $50.00
+            timestamp = LocalDateTime.now().minusDays(1),
+        )
+
+        val expense = Transaction(
+            account = account,
+            otherAccount = otherAccount,
+            type = TransactionType.EXPENSE,
+            amount = 2500L, // $25.00
+            timestamp = LocalDateTime.now(),
+        )
+
+        transactionRepository.save(income)
+        transactionRepository.save(expense)
+    }
+
+    @Test
+    fun `AC1-1 POST with valid payload returns HTTP 201 and persisted transaction JSON for income`() {
+        val initialBalance = 100.00
+        val depositAmount = 50.00
+        val (user, jwt) = createUserAndToken("user5@example.com", "Passw0rd!")
+        val account = accountRepository.save(Account(user = user, balance = (initialBalance * 100).toLong()))
+
+        val requestBody = """
+            {
+                "type": "INCOME",
+                "amount": $depositAmount,
+                "description": "Deposit"
+            }
+        """.trimIndent()
+
+        mockMvc.perform(
+            post(transactionsUrl, account.id!!)
+                .header("Authorization", jwt)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(requestBody),
+        )
+            .andExpect(status().isCreated)
+            .andExpect(jsonPath("$.type").value("INCOME"))
+            .andExpect(jsonPath("$.amount").value(depositAmount))
+            .andExpect(jsonPath("$.timestamp").exists())
+
+        // Verify account balance was updated
+        val updatedAccount = accountRepository.findById(account.id!!).get()
+        assert(updatedAccount.balance == ((initialBalance + depositAmount) * 100).toLong())
```

Comment:
No existe el `assertEquals` en el JUnit de Kotlin? Si no existe, no pasa nada, pero sí que sería más legible extraer el lado derecho de la comparación a una variable.
```suggestion
        val expectedBalance = ((initialBalance + depositAmount) * 100).toLong()
        assertEquals(expectedBalance, updatedAccount.balance)
```